src/event/reader.rs

```diff
@@ -267,9 +267,23 @@ impl EventReader {
     /// is assumed dead.
     pub async fn release_segment(&mut self, mut slice: SegmentSlice) -> Result<(), EventReaderError> {
         info!(
-            "releasing segment slice {} from reader {}",
+            "releasing segment slice {} from reader {:?}",
             slice.meta.scoped_segment, self.id
         );
+        // check if the reader is already offline.
+        if self.meta.reader_offline || !self.rg_state.lock().await.check_online(&self.id).await {
```

Comment:
This is going to add more traffic to the table. Can we combine this with the update by embedding the 'reader group offline' state into it implicitly? For example by detecting if the table is deleted.

Reply:
As per the comment of https://github.com/pravega/pravega-client-rust/blob/master/src/event/reader.rs#L360-L362, a reader could be brought offline by some other thread, should it be enough by just checking if the table is deleted? If yes, should we just send Requests::GetSegmentAttribute wirecommand to check if the table is deleted?